bin/assigner/src/state_parser.cpp

```diff
@@ -0,0 +1,94 @@
+#include "state_parser.hpp"
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/endian.hpp>
+#include <boost/endian/conversion.hpp>
+#include <cstring>
+#include <evmc/evmc.hpp>
+
+#include "state_schema.def"
+#include "vm_host.h"
+#include "zkevm_framework/json_helpers/json_helpers.hpp"
+
+// Transform hex string into Address or bytes
+template<typename OutputContainer>
+static void get_bytes(const boost::json::value hex_value, OutputContainer &result) {
+    std::vector<uint8_t> bytes = json_helpers::get_bytes(hex_value);
+    if constexpr (std::is_same_v<OutputContainer, evmc::address>) {
+        std::copy(bytes.begin(), bytes.end(), result.bytes);
+    } else if constexpr (std::is_same_v<OutputContainer, evmc::bytes32>) {
+        size_t zero_padding = 32 - bytes.size();
+        std::memset(result.bytes, 0, zero_padding);
+        std::copy(bytes.begin(), bytes.end(), result.bytes + zero_padding);
+    } else {
+        // Store the data of a transaction, could be any size
+        result.clear();
+        for (uint8_t b : bytes) {
+            result.push_back(b);
+        }
+    }
+}
+
+static void handle_storage(const boost::json::array &storage_json, evmc::account &account) {
+    auto &storage = account.storage;
+    for (const auto &item : storage_json) {
+        const boost::json::object &storage_elem = item.as_object();
+        evmc::bytes32 key, value;
+        for (const auto &elem : storage_elem) {
+            if (elem.key() == "key") {
+                get_bytes(elem.value(), key);
+            } else if (elem.key() == "value") {
+                get_bytes(elem.value(), value);
+            }
+        }
+        storage.emplace(key, value);
+    }
+}
+
+static void handle_account(const boost::json::object &account_obj, evmc::account &account) {
+    for (const auto &elem : account_obj) {
+        if (elem.key() == "balance") {
+            size_t balance_value = json_helpers::get_number(elem.value());
+            // Convert to big endian number as it required in evmc
+            boost::endian::native_to_big_inplace(balance_value);
+            // Set last bytes with balance value and others with zeros
+            constexpr size_t zero_padding = sizeof(account.balance.bytes) - sizeof(balance_value);
+            std::memcpy(account.balance.bytes + zero_padding, &balance_value,
+                        sizeof(balance_value));
+            std::memset(account.balance.bytes, 0, zero_padding);
+        } else if (elem.key() == "account") {
+            // Already processed
```

Comment:
according state_schema, account json_obj, should not contain `key = account`, or i miss something?

Reply:
Sure, sorry, it must be `address` here

Reply:
Fixed